src/main/java/com/codeborne/selenide/impl/DownloadFileToFolderCdp.java

```diff
@@ -0,0 +1,154 @@
+
+    return clickAndWaitForNewFilesInDownloadsFolder(anyClickableElement, clickable, timeout, fileFilter, action);
+  }
+
+  @CheckReturnValue
+  @Nonnull
+  private File clickAndWaitForNewFilesInDownloadsFolder(WebElementSource anyClickableElement, WebElement clickable,
+                                                        long timeout,
+                                                        FileFilter fileFilter,
+                                                        DownloadAction action) {
+    Driver driver = anyClickableElement.driver();
+    Config config = driver.config();
+    long pollingInterval = Math.max(config.pollingInterval(), 50);
+    DownloadsFolder folder = getDownloadsFolder(driver);
+
+    if (folder == null) {
+      throw new IllegalStateException("Downloads folder is not configured");
+    }
+
+    folder.cleanupBeforeDownload();
+    prepareDownloadWithCdp(driver);
+
+    long downloadStartedAt = currentTimeMillis();
+
+    action.perform(driver, clickable);
+
+    waitUntilDownloadsCompleted(timeout, pollingInterval);
+
+    Downloads newDownloads = new Downloads(folder.filesNewerThan(downloadStartedAt));
+    if (log.isInfoEnabled()) {
+      log.info("Downloaded files in {}: {}", folder, newDownloads.filesAsString());
+    }
+    if (log.isDebugEnabled()) {
+      log.debug("All downloaded files: {}", folder.filesAsString());
+    }
+
+    File downloadedFile = newDownloads.firstDownloadedFile(driver, timeout, fileFilter);
+    return archiveFile(driver, downloadedFile);
+  }
+
+  @Nullable
+  protected DownloadsFolder getDownloadsFolder(Driver driver) {
+    return driver.browserDownloadsFolder();
+  }
+
+  private void waitUntilDownloadsCompleted(long timeout, long pollingInterval) {
+    for (long start = currentTimeMillis(); currentTimeMillis() - start < timeout; pause(pollingInterval)) {
+      if (downloadComplete.get()) {
+        log.info("File {} download is complete", fileName.get());
+        return;
+      }
+    }
+  }
+
+  private void pause(long milliseconds) {
+    try {
+      sleep(milliseconds);
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Nonnull
+  protected File archiveFile(Driver driver, File downloadedFile) {
+    File uniqueFolder = downloader.prepareTargetFolder(driver.config());
+    File archivedFile = new File(uniqueFolder, downloadedFile.getName());
+    moveFile(downloadedFile, archivedFile);
+    log.debug("Moved the downloaded file {} to {}", downloadedFile, archivedFile);
+    return archivedFile;
+  }
+
+  private void initDevTools(Driver driver) {
+    if (driver.browser().isChromium()) {
+      devTools = ((HasDevTools) driver.getWebDriver()).getDevTools();
+      devTools.createSession();
+      devTools.send(Page.enable());
+    } else {
+        throw new RuntimeException("The browser that you choose doesn't have such functionality.");
```

Comment:
throw new RuntimeException("The browser you selected <ADD_BROWSER_NAME_HERE> doesn't have Chrome Devtools protocol functionality.");

Reply:
Message has been changed